airflow/config_templates/config.yml

```
@@ -2320,6 +2320,16 @@ scheduler:
       type: string
       example: ~
       default: "^[A-Za-z0-9_.~:+-]+$"
+    legacy_cron_data_intervals:
+      description: |
+        Which Timetable to use when a cron string is provided to `schedule` argument of
+        a DAG. If True, CronDataIntervalTimetable is used, which is the legacy Airflow behavior suitable
+        for DAGs with well-defined data_interval. If False, CronTriggerTimetable is used,
+        which is closer to the behavior of cron itself.
```

```suggestion
    create_cron_intervals:
      description: |
        Whether to create DAG runs that span an interval or one single point in time for cron schedules. If
        True, you get contiguous intervals from the end of the previous interval up to the scheduled datetime.
        This internally uses the CronDataIntervalTimetable. If False, you get intervals with a start and end
        set to the same scheduled datetime. This internally uses the CronTriggerTimetable.
```

I suggest giving the config a name and description that explains the effect on the interval -- to me "legacy Airflow behavior" doesn't tell what the difference in behavior is.